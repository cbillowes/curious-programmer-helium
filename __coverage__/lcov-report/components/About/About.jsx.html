<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/About/About.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/About</a> About.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/9</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >import React, { Component } from "react"</span>
<span class="cstat-no" title="statement not covered" >import "./About.scss"</span>
&nbsp;
function <span class="fstat-no" title="function not covered" >Bio() {</span>
<span class="cstat-no" title="statement not covered" >  return (</span>
    &lt;div className="profile"&gt;
      &lt;div className="photo" /&gt;
      &lt;h1 className="not-for-print"&gt;Hello Clarice&lt;/h1&gt;
      &lt;div className="for-print"&gt;
        &lt;h1&gt;Clarice Bouwer&lt;/h1&gt;
        &lt;h2&gt;Senior Software Developer&lt;/h2&gt;
      &lt;/div&gt;
      &lt;p&gt;
        I write code. I share code. I love code. I am curious and fascinated about computers.   
      &lt;/p&gt;
      &lt;p&gt;
        When I am not coding I am making people smile. I make jokes and giggle offering a warm radiant energy that people seem to love.
        This makes me happy.
      &lt;/p&gt;
    &lt;/div&gt;
  )
}
&nbsp;
class About extends Component {
<span class="fstat-no" title="function not covered" >  constructor</span>(props) <span class="cstat-no" title="statement not covered" >{</span>
<span class="cstat-no" title="statement not covered" >    super(props)</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    this.state = {</span>
      elaborated: false
    }
  }
&nbsp;
  isElaborate = <span class="fstat-no" title="function not covered" >() =&gt; </span>{
<span class="cstat-no" title="statement not covered" >    return this.state.elaborated</span>
  }
&nbsp;
  toggle = <span class="fstat-no" title="function not covered" >() =&gt; </span>{
<span class="cstat-no" title="statement not covered" >    this.setState({ elaborated: !this.isElaborate() })</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  render</span>() {
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;div className="about"&gt;
        &lt;Bio /&gt;
        &lt;div className="read-more not-for-print"&gt;
          &lt;div&gt;
            &lt;div className="button" onClick={this.toggle}&gt;
              {this.isElaborate() ? "show less" : "read more"}
            &lt;/div&gt;
          &lt;/div&gt;
          &lt;aside
            className={`elaborate ${this.isElaborate() ? "elaborated" : ""}`}
          &gt;
            &lt;p&gt;
              I have an eye for aesthetically pleasing, user-friendly, developer-friendly
              products and solutions. My skills are not limited to web development. I offer 
              graphical, quality assurance and operational skills by experience.
            &lt;/p&gt;
            &lt;p&gt;
              I work well within teams and offer intrinsic leadership skills and 
              encourage others to voice their opinions.
            &lt;/p&gt;
            &lt;p&gt;
              I find great happiness in coding, learning and solving problems. I also like 
              to collect e-books, read, watch anime, listen to punk rock music and 
              try to push myself outside of my comfort zone.
            &lt;/p&gt;
          &lt;/aside&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    )
  }
}
&nbsp;
export default About
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 09:15:22 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
