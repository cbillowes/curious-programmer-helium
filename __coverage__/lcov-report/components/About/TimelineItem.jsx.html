<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/About/TimelineItem.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/About</a> TimelineItem.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/11</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >import React, { Component } from "react"</span>
<span class="cstat-no" title="statement not covered" >import Link from "gatsby-link"</span>
<span class="cstat-no" title="statement not covered" >import External from "../Link/ExternalLink"</span>
&nbsp;
class TimelineItem extends Component <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >{</span></span>
  getCompanyHeading = <span class="fstat-no" title="function not covered" >(c</span>ompany, companyLink) =&gt; {
<span class="cstat-no" title="statement not covered" >    if (companyLink) {</span>
<span class="cstat-no" title="statement not covered" >      if (companyLink.startsWith("http")) {</span>
<span class="cstat-no" title="statement not covered" >        return &lt;External to={companyLink}&gt;{company}&lt;/External&gt;</span>
      } else {
<span class="cstat-no" title="statement not covered" >        return &lt;Link to={companyLink}&gt;{company}&lt;/Link&gt;</span>
      }
    }
<span class="cstat-no" title="statement not covered" >    return company</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  render</span>() {
    const {
      type,
      company,
      companyLink,
      duration,
      title
    } = <span class="cstat-no" title="statement not covered" >this.props</span>
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;li className={`item ${type}`}&gt;
        &lt;section&gt;
          &lt;div className="meta"&gt;
            &lt;h2&gt;{this.getCompanyHeading(company, companyLink)}&lt;/h2&gt;
            &lt;span&gt;{duration}&lt;/span&gt;
          &lt;/div&gt;
          &lt;div className="icon"&gt;
            &lt;span /&gt;
          &lt;/div&gt;
          &lt;div className="description"&gt;
            &lt;h3&gt;{title}&lt;/h3&gt;
            {this.props.children}
          &lt;/div&gt;
        &lt;/section&gt;
      &lt;/li&gt;
    )
  }
}
&nbsp;
export default TimelineItem</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 09:15:22 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
