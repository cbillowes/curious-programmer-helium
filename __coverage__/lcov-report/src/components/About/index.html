<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/About</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/About
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">65.28% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>47/72</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">68.75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>33/48</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">58.06% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>18/31</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">63.77% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>44/69</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="About.jsx"><a href="About.jsx.html">About.jsx</a></td>
	<td data-value="83.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 83%;"></div><div class="cover-empty" style="width:17%;"></div></div></td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	</tr>

<tr>
	<td class="file high" data-value="Contact.jsx"><a href="Contact.jsx.html">Contact.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file high" data-value="Technologies.jsx"><a href="Technologies.jsx.html">Technologies.jsx</a></td>
	<td data-value="93.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 93%;"></div><div class="cover-empty" style="width:7%;"></div></div></td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="30" class="abs high">28/30</td>
	<td data-value="86.36" class="pct high">86.36%</td>
	<td data-value="22" class="abs high">19/22</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="92.59" class="pct high">92.59%</td>
	<td data-value="27" class="abs high">25/27</td>
	</tr>

<tr>
	<td class="file low" data-value="Timeline.jsx"><a href="Timeline.jsx.html">Timeline.jsx</a></td>
	<td data-value="12" class="pic low"><div class="chart"><div class="cover-fill" style="width: 12%;"></div><div class="cover-empty" style="width:88%;"></div></div></td>
	<td data-value="12" class="pct low">12%</td>
	<td data-value="25" class="abs low">3/25</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="15.38" class="pct low">15.38%</td>
	<td data-value="13" class="abs low">2/13</td>
	<td data-value="12" class="pct low">12%</td>
	<td data-value="25" class="abs low">3/25</td>
	</tr>

<tr>
	<td class="file high" data-value="TimelineItem.jsx"><a href="TimelineItem.jsx.html">TimelineItem.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="TimelineLegend.jsx"><a href="TimelineLegend.jsx.html">TimelineLegend.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="16" class="abs medium">8/16</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 14:17:11 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
