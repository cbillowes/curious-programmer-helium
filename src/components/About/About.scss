@import "../../layout/variables.scss";

$profile-size: 300px;

.about {
    text-align: center;

    @include medium-media {
        text-align: left;
    }


    @include small-media {
        text-align: left;
    }


    .profile {
        max-width: $container-width;
        margin: 0 auto;

        @include medium-media {
            margin-left: 4em;
        }


        @include small-media {
            margin-left: 4em;
        }


        h1 {
            font-size: $font-size-lg-h1 + 1;
            margin-top: 10px;
            margin-bottom: 0;

            @include medium-media {
                font-size: $font-size-md-h1 + 0.75;
                text-align: center;
            }


            @include small-media {
                font-size: $font-size-sm-h1 + 0.5;
                text-align: center;
            }
        }

        .photo {
            width: 300px;
            height: 300px;
            border-radius: 50%;
            background-image: url("/images/profile.jpg");
            background-position: center;
            background-size: 300px;
            margin: 0 auto;

            @include small-media {
                width: $profile-size / 2;
                height: $profile-size / 2;
                background-size: $profile-size / 2;
            }
        }
    }

    .read-more {
        text-align: center;

        .button {
            @include button;

            background-color: $foreground-lighter;
        }
    }

    .elaborate {
        display: none;

        &.elaborated {
            display: block;
        }
    }

    .elaborated {
        display: block;
        margin: 1em 2em;

        @media screen and (min-width: 801px) {
            @include highlighted-section;
        }

        @include medium-media {
            margin: 0.5em;
            text-align: left;
        }


        @include small-media {
            text-align: left;
            margin: 0.5em;
        }
    }
}

@media print {
    .about {
        .profile {
            text-align: left;

            h1 {
                font-size: 2.5em;
            }

            .photo {
                float: right;
                width: 200px;
                height: 200px;
            }
        }
    }
}
