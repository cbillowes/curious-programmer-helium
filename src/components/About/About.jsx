import React, { Component } from "react"
import "./About.scss"

function Bio() {
  return (
    <div className="profile">
      <div className="photo" />
      <h1 className="not-for-print">Hello Clarice</h1>
      <div className="for-print">
        <h1>Clarice Bouwer</h1>
        <h2>Senior Software Developer</h2>
      </div>
      <p>
        I write code. I share code. I love code. I am curious and fascinated about computers.   
      </p>
      <p>
        When I am not coding I am making people smile. I make jokes and giggle offering a warm radiant energy that people seem to love.
        This makes me happy.
      </p>
    </div>
  )
}

class About extends Component {
  constructor(props) {
    super(props)

    this.state = {
      elaborated: false
    }
  }

  isElaborate = () => {
    return this.state.elaborated
  }

  toggle = () => {
    this.setState({ elaborated: !this.isElaborate() })
  }

  render() {
    return (
      <div className="about">
        <Bio />
        <div className="read-more not-for-print">
          <div>
            <div className="button" onClick={this.toggle}>
              {this.isElaborate() ? "show less" : "read more"}
            </div>
          </div>
          <aside
            className={`elaborate ${this.isElaborate() ? "elaborated" : ""}`}
          >
            <p>
              I have an eye for aesthetically pleasing, user-friendly, developer-friendly
              products and solutions. My skills are not limited to web development. I offer 
              graphical, quality assurance and operational skills by experience.
            </p>
            <p>
              I work well within teams and offer intrinsic leadership skills and 
              encourage others to voice their opinions.
            </p>
            <p>
              I find great happiness in coding, learning and solving problems. I also like 
              to collect e-books, read, watch anime, listen to punk rock music and 
              try to push myself outside of my comfort zone.
            </p>
          </aside>
        </div>
      </div>
    )
  }
}

export default About
