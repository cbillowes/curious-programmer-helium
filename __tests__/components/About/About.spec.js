import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import About from "../../../src/components/About/About"

describe("an about component", () => {
  it("should render", () => {
    const component = <About />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render contents", () => {
    const component = mount(<About />)
    expect(component.find(".bio")).toBeDefined()
    expect(component.find(".profile")).toBeDefined()
  })
})