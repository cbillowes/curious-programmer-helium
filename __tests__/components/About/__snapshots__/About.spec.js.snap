// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an about component should render 1`] = `
<div
  className="about"
>
  <div
    className="profile"
  >
    <div
      className="photo"
    />
    <h1
      className="not-for-print"
    >
      Hello Clarice
    </h1>
    <div
      className="for-print"
    >
      <h1>
        Clarice Bouwer
      </h1>
      <h2>
        Senior Software Developer
      </h2>
    </div>
    <p>
      I write code. I share code. I love code. I am curious and fascinated about computers.
    </p>
    <p>
      When I am not coding I am making people smile. I make jokes and giggle offering a warm radiant energy that people seem to love. This makes me happy.
    </p>
  </div>
  <div
    className="read-more not-for-print"
  >
    <div>
      <div
        className="button"
        onClick={[Function]}
      >
        read more
      </div>
    </div>
    <aside
      className="elaborate "
    >
      <p>
        I have an eye for aesthetically pleasing, user-friendly, developer-friendly products and solutions. My skills are not limited to web development. I offer graphical, quality assurance and operational skills by experience.
      </p>
      <p>
        I work well within teams and offer intrinsic leadership skills and encourage others to voice their opinions.
      </p>
      <p>
        I find great happiness in coding, learning and solving problems. I also like to collect e-books, read, watch anime, listen to punk rock music and try to push myself outside of my comfort zone.
      </p>
    </aside>
  </div>
</div>
`;
